worlds/sc2wol/MissionTables.py

```diff
@@ -19,13 +26,21 @@ class MissionPools(IntEnum):
     FINAL = 4
 
 
+mission_pool_names = ['Starter', 'Easy', 'Medium', 'Hard', 'Final']
+
+
+class MissionInfoUiFlags(Flag):
+    PrependSpacer = auto()
+
+
 class MissionInfo(NamedTuple):
     id: int
-    required_world: List[int]
+    required_world: Union[List[int], List[str]]
     category: str
     number: int = 0  # number of worlds need beaten
     completion_critical: bool = False  # missions needed to beat game
     or_requirements: bool = False  # true if the requirements should be or-ed instead of and-ed
+    ui_flags: int = 0  # for sending flags to the UI. Uses MissionInfoUiFlags, but stored as an int so pickling works
```

Comment:
Thinking about this overnight, I realized it's probably easier and better to give this structure a version number rather than checking types / length everywhere. Length == 6 would imply version 0, this would be version 1, and types could be assumed based on the version.

The reason for the version number is for client backwards-compatibility with games that were generated with older versions of the apworld. It's probably less important now that many changes have merged to main for 0.4.3, but I think we're still maintaining that backwards compatibility and it can be a lot more seamless if we use a version number to maintain client compatibility with at least a small range of apworld versions.

Ideally, `required_world` would be entirely based on string keys or mission IDs rather than order it was placed in a dict, though.